_source/_posts/2022-01-05-eleventy.md

```diff
@@ -0,0 +1,375 @@
+      </div>
+    </header>
+
+    ${data.content}
+        
+        `;
+};
+```
+
+This file extends the base layout you just created by creating a header with a `Home` button and a `Login` button that will direct the user to the `Login` page with your Okta sign-in widget.
+
+Finally, add a new file called `_layout_aunthenticated.11ty.js`.  Add the code below to it.
+
+```javascript
+exports.data = {
+title: "11ty And Okta",
+layout: "_layout_base.11ty.js",
+};
+
+exports.render = function (data) {
+return `
+
+<header
+    class="d-flex flex-wrap align-items-center justify-content-center justify-content-md-between py-3 mb-4 border-bottom">
+    <a href="/" class="d-flex align-items-center col-md-3 mb-2 mb-md-0 text-dark text-decoration-none">
+        Home
+    </a>
+
+    <div class="col-md-3 text-end">
+        <button type="button" class="btn btn-outline-primary me-2" onclick='logout()'>Logout</button>
+    </div>
+</header>
+
+${data.content}
+
+<script type="text/javascript">
+    function logout() {
+        oktaSignIn.authClient.signOut();
+        location.reload();
+    }
+
+    if(!oktaSignIn?.authClient?.token) {
+        window.location='/Login';
+    }
+
+    oktaSignIn.authClient.token.getUserInfo().then(function(user) {
+        if(loadPage) {
+            loadPage();
+        }
+    }).catch(function(e) {
+        console.log(e);
+        window.location='/Login'
+    });
+</script>
+`;
+};
+```
+
+Again, this file will extend the base layout.  It also adds some additional functionality to each page using this out.  When the page loads it will attempt to contact the Okta authorization server to obtain information about the user with the id token stored in the browser.  If that request is successful it will call the `loadPage` function implemented in those pages.  Otherwise, it redirects the user to the `Login` page.  This page also contains a header with a `Logout` button and a method for `logout` that uses the sign-in widget to log the user out.
+
+## Start your site
+
+If you are still running the site then 11ty would have hot reloaded the site and you should see the home page.  If not, run the command `npx @11ty/eleventy --serve`.  The command will build your templates into static pages and then serve the site on `localhost:8080`.  Navigate to `http://localhost:8080` to see the home page.
+
+{% img blog/eleventy/Home Page.png alt:"Home Page" width:"800" %}{: .center-image }
+
+Next, click the login button to see the Okta Sign-in widget in action.  You can log in with your Okta credentials and see the protected page.
+
+{% img blog/eleventy/Login page.png alt:"Login Page" width:"800" %}{: .center-image }
+
+## Learn more about working with 11ty
+
+11ty is a great tool for building practical static websites using several different templating languages.  Together with Okta you can create secure static sites easily and quickly.  
```

Comment:
Can we add a bit of a summary of what the readers learned in the post? :)

Reply:
added